src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -8118,29 +8133,63 @@ void checkWellKnown(WellKnownMember member)
 
             var lookupResult = LookupResult.GetInstance();
 
-            if (TryLookupLengthOrCount(receiverType, lookupResult, out lengthOrCountProperty, ref useSiteInfo) &&
-                TryFindIndexOrRangeIndexerPattern(lookupResult, receiverOpt, receiverType, argIsIndex, out patternSymbol, diagnostics, ref useSiteInfo))
+            if (TryBindLengthOrCount(syntax, receiverOpt, receiverType, out lengthOrCountAccess, diagnostics) &&
+                TryFindIndexOrRangeImplicitIndexer(syntax, lookupResult, receiverOpt, receiverType, argIsIndex, out implicitIndexerAccess, out argumentPlaceholders, diagnostics))
+            {
+                CheckValue(lengthOrCountAccess, BindValueKind.RValue, diagnostics);
+
+                lookupResult.Free();
+                return true;
+            }
+
+            lengthOrCountAccess = null;
+            implicitIndexerAccess = null;
+            argumentPlaceholders = default;
+            lookupResult.Free();
+            return false;
+        }
+
+        private bool TryBindLengthOrCount(
+            SyntaxNode syntax,
+            BoundExpression? receiverOpt,
+            TypeSymbol receiverType,
+            [NotNullWhen(true)] out BoundExpression? lengthOrCountAccess,
+            BindingDiagnosticBag diagnostics)
+        {
+            var lookupResult = LookupResult.GetInstance();
+
+            if (TryLookupLengthOrCount(syntax, receiverType, lookupResult, out var lengthOrCountProperty, diagnostics))
             {
-                CheckImplicitThisCopyInReadOnlyMember(receiverOpt, lengthOrCountProperty.GetMethod, diagnostics);
+                lengthOrCountAccess = BindPropertyAccess(syntax, receiverOpt, lengthOrCountProperty, diagnostics, lookupResult.Kind, hasErrors: false);
+                ReportDiagnosticsIfObsolete(diagnostics, lengthOrCountProperty, syntax, hasBaseReceiver: false);
```

Comment:
>ReportDiagnosticsIfObsolete(diagnostics, lengthOrCountProperty, syntax, hasBaseReceiver: false); [](http://example.com/codeflow?start=16&length=96)

I think this check should be moved to the ```BindPropertyAccess``` method and its callers should be adjusted accordingly.

Reply:
I left the obsolete check (`ReportDiagnosticsIfObsolete(diagnostics, lengthOrCountProperty, syntax, hasBaseReceiver: false);`) in `TryBindLengthOrCount` because of `hasBaseReceiver`.
I can file a follow-up issue. Let me know.